### DISCUSSION TITLE
How to temporarily stop an OSD process launched by Rook?

### DISCUSSION THREAD

**Discussion by dharmateja** [rook/rook#10091] [Q&A]
Hey all. 

I am new to Ceph and have launched a ceph cluster using rook. I am trying to examine the contents of an OSD data directory and there are a few ceph tools that allow me to export the bluestore data to an output directory (ceph-bluestore-tool, eph-objectstore-tool etc). As I tried to run these tools, they complain about not being able to get a lock because the OSD process likely has the lock.

`
bluestore(/var/lib/ceph/osd/ceph-4) _lock_fsid failed to lock /var/lib/ceph/osd/ceph-4/fsid (is another ceph-osd still running?)(11) Resource temporarily unavailable
`

The ceph documentation has often pointed to using `systemctl stop ceph-osd@<id>` and when I try that on a cluster that was setup with rook, I see the following error. 

`
sh-4.4# sudo systemctl stop ceph-osd@4
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
sh-4.4#
`

I assume this is because the OSD container was launched by rook and and not with systemd. Is there a way to temporarily pause an OSD process/container that gives me enough time to export the data out of the container?

**Comment by travisn** [comment]
There isn’t currently a way to easily pause the OSDs in rook before they start. You can try the following:
1. Scale down the rook operator so it doesn’t reconcile temporarily
2. Modify the osd deployment spec so the main container is just a pause container instead of running the ceph-osd command. The init containers should remain the same so they still initialize the osd container.
3. Connect to the pause container and run your commands
4. Start the operator again and it will restore the osd daemon container

**Comment by travisn** [comment]
Also, what bluestore data are you trying to export? If there is a general utility, perhaps Rook could add another init container that dumps the info.
